cmd/rendmp/src/lib.rs

```diff
@@ -1002,55 +1047,959 @@ fn rendmp_open_pin(
+        self.calls.push(Call::ReadWord(0));
+        Ok(())
+    }
+
+    fn write_word(
+        &mut self,
+        index: u32,
+        has_rail: bool,
+        op: u8,
+        data: u16,
+    ) -> Result<()> {
+        let payload = self.write_word.payload(&[
+            ("index", index.into()),
+            ("has_rail", has_rail.into()),
+            ("op", op.into()),
+            ("data", data.into()),
+        ])?;
+        self.context.idol_call_ops(
+            &self.write_word,
+            &payload,
+            &mut self.ops,
+        )?;
+        self.calls.push(Call::WriteWord);
+        Ok(())
+    }
+
+    fn read_byte(&mut self, index: u32, has_rail: bool, op: u8) -> Result<()> {
+        let payload = self.read_byte.payload(&[
+            ("index", index.into()),
+            ("has_rail", has_rail.into()),
+            ("op", op.into()),
+        ])?;
+        self.context.idol_call_ops(&self.read_byte, &payload, &mut self.ops)?;
+        self.calls.push(Call::ReadByte(0));
+        Ok(())
+    }
+
+    fn write_byte(
+        &mut self,
+        index: u32,
+        has_rail: bool,
+        op: u8,
+        data: u8,
+    ) -> Result<()> {
+        let payload = self.write_byte.payload(&[
+            ("index", index.into()),
+            ("has_rail", has_rail.into()),
+            ("op", op.into()),
+            ("data", data.into()),
+        ])?;
+        self.context.idol_call_ops(
+            &self.write_byte,
+            &payload,
+            &mut self.ops,
+        )?;
+        self.calls.push(Call::WriteByte);
+        Ok(())
+    }
+
+    fn read_word32(
+        &mut self,
+        index: u32,
+        has_rail: bool,
+        op: u8,
+    ) -> Result<()> {
+        let payload = self.read_word32.payload(&[
+            ("index", index.into()),
+            ("has_rail", has_rail.into()),
+            ("op", op.into()),
+        ])?;
+        self.context.idol_call_ops(
+            &self.read_word32,
+            &payload,
+            &mut self.ops,
+        )?;
+        self.calls.push(Call::ReadWord32(0));
+        Ok(())
+    }
+
+    fn write_word32(
+        &mut self,
+        index: u32,
+        has_rail: bool,
+        op: u8,
+        data: u32,
+    ) -> Result<()> {
+        let payload = self.write_word32.payload(&[
+            ("index", index.into()),
+            ("has_rail", has_rail.into()),
+            ("op", op.into()),
+            ("data", data.into()),
+        ])?;
+        self.context.idol_call_ops(
+            &self.write_word32,
+            &payload,
+            &mut self.ops,
+        )?;
+        self.calls.push(Call::WriteWord32);
+        Ok(())
+    }
+
+    fn run(
+        &mut self,
+        core: &mut dyn humility::core::Core,
+    ) -> Result<Vec<Result<Call, String>>> {
+        let mut ops = std::mem::take(&mut self.ops);
+        ops.push(Op::Done);
+        let r = self.context.run(core, &ops, None)?;
+        // TODO: decode error messages here
```

Comment:
Did you want to address this TODO before merging?

Reply:
This is actually done in the loop + match below, so I've removed the comment in 9c0d4b6125ed915d839915abd98bdeadab30e8c0